comfy_panel/special_games.lua

```diff
@@ -0,0 +1,199 @@
+local Event = require 'utils.event'
+local Color = require 'utils.color_presets'
+local valid_special_games = {
+	turtle = {
+		name = "Turtle",
+		config = {
+			[1] = {name = "label1", type = "label", caption = "moat width"},
+			[2] = {name = 'moat_width', type = "textfield", text = "3", numeric = true, width = 40},
+			[3] = {name = "label2", type = "label", caption = "entrance width"},
+			[4] = {name = 'entrance_width', type = "textfield", text = "20", numeric = true, width = 40},
+			[5] = {name = "label3", type = "label", caption = "size x"},
+			[6] = {name = 'size_x', type = "textfield", text = "200", numeric = true, width = 40},
+			[7] = {name = "label4", type = "label", caption = "size y"},
+			[8] = {name = 'size_y', type = "textfield", text = "200", numeric = true, width = 40}
+
+		},
+		button = {name = "turtle_apply", type = "button", caption = "Apply"}
+	},
+	infinity_chest = {
+		name = "Infinity chest",
+		config = {
+			[1] = {name = "separate_chests", type = "checkbox", caption = "Separate chest for each item", state = false},
+			[2] = {name = "eq1", type = "choose-elem-button", elem_type = "item"},
+			[3] = {name = "eq2", type = "choose-elem-button", elem_type = "item"},
+			[4] = {name = "eq3", type = "choose-elem-button", elem_type = "item"},
+			[5] = {name = "eq4", type = "choose-elem-button", elem_type = "item"},
+			[6] = {name = "eq5", type = "choose-elem-button", elem_type = "item"}
+		},
+		button = {name = "infinity_chest_apply", type = "button", caption = "Apply"}
+	}
+}
+
+local function generate_turtle(moat_width, entrance_width, size_x, size_y)
+	game.print("Special game turtle is being generated!", Color.warning)
+	local surface = game.surfaces[global.bb_surface_name]
```

Comment:
First thing before making any moat generation, you need to make sure you chart the work surface, else you will see artifacts in moat sometimes randomly.:
```suggestion
	local surface = game.surfaces[global.bb_surface_name]
	game.forces["spectator"].chart(surface, {{-size_x/2-moat_width, -size_y-moat_width}, {size_x/2+moat_width, size_y+moat_width}})
```

Reply:
The only artifacts I've spotted were remnants, and I didn't think we would care about them :D

Reply:
Even so, its nice to have discovery of the area that includes moat.